server/runtime/src/main/java/org/infinispan/server/network/NetworkAddress.java

```diff
@@ -97,6 +131,11 @@ public static NetworkAddress matchHost(String name, String regex) throws IOExcep
       return new NetworkAddress(name, findAddress(a -> a.getHostName().matches(regex)));
    }
 
+   public static NetworkAddress match(String name, Predicate<NetworkInterface> ifMatcher, Predicate<InetAddress> addressMatcher) throws IOException {
+      NetworkInterface networkInterface = findInterface(ifMatcher);
+      return new NetworkAddress(name, findAddress(networkInterface, addressMatcher));
```

Comment:
`findAddress()` may return `null`. 
you may want to `throw new IOException("No matching addresses found");` if that is the case.